doc/application/application.rst

```diff
@@ -966,50 +966,60 @@ preferred, since it correctly handles dependencies between options.
 Device Tree Overlays
 ====================
 
-Zephyr uses Device Tree to describe the hardware it runs on. Sometimes it is
-necessary to modify the definitions in the :file:`.dts` files provided in the
-board folder in order to customize settings for a particular application.
-For more information on Device Tree see :ref:`device-tree`.
+As described in :ref:`device-tree`, Zephyr uses Device Tree to
+describe the hardware it runs on. This section describes how you can
+modify an application build's device tree using overlay files.
 
-The Zephyr build system takes a Device Tree configuration
-option's value from the first source in which it is specified, taken from the
-following available sources, in order:
+Overlay files, which customarily have the :file:`.overlay` extension,
+contain device tree fragments which add to or modify the device tree
+used while building a Zephyr application. To add an overlay file or
+files to the build, set the CMake variable :makevar:`DTC_OVERLAY_FILE`
+to a whitespace-separated list of your overlay files.
 
-#. An application's current DT configuration (i.e. the file
-   :file:`zephyr/<board>.dts.pre.tmp` in the build directory).
+The Zephyr build system begins creation of a device tree by running
+the C preprocessor on a file which includes the following:
 
-#. The application's DT configuration file(s) given by the
-   :makevar:`DTC_OVERLAY_FILE` variable, either as set explicitly by the user
-   or using one of the default values detailed below.
+#. Configuration options from :ref:`Kconfig <configuration>`.
 
-#. Common overlays conditionally included by :file:`dts/common/common.dts`.
+#. The board's device tree source file, which by default is the Zephyr
+   file :file:`boards/ARCHITECTURE/BOARD/BOARD.dts`. (This location
```

Comment:
Since it's not actually a folder called ARCHITECTURE or BOARD, it might be better to write this as:

    :file:`boards/<ARCHITECTURE>/<BOARD>/<BOARD>.dts`